.github/workflows/build-ffmpeg.yml

```diff
@@ -10,160 +10,239 @@ on:
   workflow_dispatch:
 
 jobs:
-  linux:
-    name: "Linux"
-    runs-on: ubuntu-22.04
+  ffmpeg:
+    name: "FFmpeg"
     # https://trac.ffmpeg.org/wiki/CompilationGuide/Ubuntu
+    # https://trac.ffmpeg.org/wiki/CompilationGuide/macOS
+    # https://trac.ffmpeg.org/wiki/CompilationGuide/MinGW
+    strategy:
+      fail-fast: false  # false to test all, true to fail entire job if any fail
+      matrix:
+        include:
+          - os_type: linux
+            os: ubuntu-22.04
+            shell: bash
+            cmake_generator: Unix
+            ffmpeg_extras: |
+              --enable-amf \
+              --enable-encoder=h264_amf \
+              --enable-encoder=h264_nvenc,hevc_nvenc \
+              --enable-encoder=h264_vaapi,hevc_vaapi \
+              --enable-encoder=libx264,libx265 \
+              --enable-nvenc \
+              --enable-vaapi
+          - os_type: macos
+            os: macos-11
+            shell: bash
+            cmake_generator: Unix
+            ffmpeg_extras: |
+              --enable-encoder=libx264,libx265 \
+              --enable-encoder=h264_videotoolbox,hevc_videotoolbox \
+              --enable-videotoolbox
+          - os_type: windows
+            os: windows-2022
+            shell: msys2 {0}
+            cmake_generator: MSYS
+            ffmpeg_extras: |
+              --enable-amf \
+              --enable-encoder=h264_amf,hevc_amd \
+              --enable-encoder=libx264,libx265 \
+              --enable-encoder=h264_mf,hevc_mf \
+              --enable-encoder=h264_nvenc,hevc_nvenc \
+              --enable-nvenc \
+              --enable-mediafoundation
```

Comment:
N.B. this will require all or some combination of `mfplat ole32 strmiids mfuuid` added to link-time libraries on Windows.